components/dashboard/src/components/Modal.tsx

```diff
@@ -1,24 +1,20 @@
-import { useState } from "react";
-
 export default function Modal(props: {
     children: React.ReactChild[] | React.ReactChild,
     visible: boolean,
     closeable?: boolean,
-    onClose?: () => void
+    onClose: () => void
 }) {
-    const [visible, setVisible] = useState(props.visible);
-    const hide = () => setVisible(false);
-    if (!visible) {
+    if (!props.visible) {
         return null;
     }
+    setTimeout(() => window.addEventListener('click', props.onClose, {once: true}), 0);
     return (
-        <div className="absolute top-0 left-0 bg-black bg-opacity-40 z-50 w-screen h-screen" >
+        <div className="fixed top-0 -left-2 bg-black bg-opacity-70 z-50 w-screen h-screen" >
             <div className="bg-transparent h-1/3" />
-            <div className="bg-white rounded-md px-6 py-4 max-w-lg mx-auto">
+            <div className="bg-white border rounded-lg px-6 py-4 max-w-lg mx-auto">
```

Comment:
@gtsiolis said modal padding is 24px (p-6):

```suggestion
            <div className="bg-white border rounded-lg p-4 max-w-lg mx-auto">
```